homeassistant/components/deconz/deconz_device.py

```diff
@@ -15,32 +17,42 @@
 from .const import DOMAIN as DECONZ_DOMAIN
 from .gateway import DeconzGateway
 
+_DeviceTypeT = TypeVar(
+    "_DeviceTypeT",
+    bound=Union[
+        PydeconzGroup,
+        PydeconzLightBase,
+        PydeconzSensorBase,
+        PydeconzScene,
+    ],
+)
+
 
-class DeconzBase:
+class DeconzBase(Generic[_DeviceTypeT]):
     """Common base for deconz entities and events."""
 
     def __init__(
         self,
-        device: DeconzGroup | DeconzLight | DeconzSensor | PydeconzScene,
+        device: _DeviceTypeT,
         gateway: DeconzGateway,
     ) -> None:
         """Set up device and add update callback to get data from websocket."""
-        self._device = device
+        self._device: _DeviceTypeT = device
         self.gateway = gateway
 
     @property
     def unique_id(self) -> str:
         """Return a unique identifier for this device."""
         assert not isinstance(self._device, PydeconzScene)
```

Comment:
```suggestion
        assert isinstance(self._device, PydeconzDevice)
```
Instead of adding the `type: ignores`, try doing a positive `assert` with the base class.